subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -2064,3 +2067,61 @@ void bt_mesh_cfg_cli_timeout_set(int32_t timeout)
 {
 	msg_timeout = timeout;
 }
+
+int bt_mesh_comp_page_0_get(struct bt_mesh_comp_page_0 *page,
+			    struct net_buf_simple *buf)
+{
+	if (buf->len < 10) {
+		return -EINVAL;
+	}
+
+	page->cid = net_buf_simple_pull_le16(buf);
+	page->pid = net_buf_simple_pull_le16(buf);
+	page->vid = net_buf_simple_pull_le16(buf);
+	page->crpl = net_buf_simple_pull_le16(buf);
+	page->feat = net_buf_simple_pull_le16(buf);
+	page->_buf = buf;
+
+	return 0;
+}
+
+int bt_mesh_comp_elem_pull(const struct bt_mesh_comp_page_0 *page,
+			   struct bt_mesh_comp_elem *elem)
+{
+	size_t modlist_size;
+
+	if (page->_buf->len < 4) {
+		return -ENOENT;
+	}
+
+	elem->loc = net_buf_simple_pull_le16(page->_buf);
+	elem->nsig = net_buf_simple_pull_u8(page->_buf);
+	elem->nvnd = net_buf_simple_pull_u8(page->_buf);
+
+	modlist_size = elem->nsig * 2 + elem->nvnd * 4;
+
+	if (page->_buf->len < modlist_size) {
+		return -EINVAL;
+	}
+
+	elem->_buf =
+		(uint8_t *)net_buf_simple_pull_mem(page->_buf, modlist_size);
+	return 0;
+}
+
+uint16_t bt_mesh_comp_elem_mod(struct bt_mesh_comp_elem *elem, int idx)
+{
+	return sys_get_le16(&elem->_buf[idx * 2]);
```

Comment:
Should you check elem->nsig first?

Reply:
Added a bounds check, but there's no room for an error code in the return types of these functions, so I have made them return `0xffff`. This can also be a valid return value though, so I think it's the lesser of two evils. I'd prefer not to change the return type, as this would mess up the vnd function.